### DISCUSSION TITLE
Descriptions are not generated in the form of comments

### DISCUSSION THREAD

**Discussion by nandita1** [dotansimha/graphql-code-generator#8357] [Q&A] [answered]
When we add the schema as a `*.graphql` file, descriptions of fields inside the schema file do not get added as JSDOC comments in the generated types file but they do get generate if the schema is a `*.json` file.
With `*.graphql` schema file:
<img width="529" alt="Screenshot 2022-09-13 at 8 43 06 PM" src="https://user-images.githubusercontent.com/36603774/189939903-5683b71f-7a6d-4152-801c-ab4f64b9b33e.png">
<img width="522" alt="Screenshot 2022-09-13 at 8 44 35 PM" src="https://user-images.githubusercontent.com/36603774/189939926-77a76697-c4b5-48d4-948f-50387379671c.png">
With `*.json` schema file:
<img width="515" alt="Screenshot 2022-09-13 at 8 43 22 PM" src="https://user-images.githubusercontent.com/36603774/189940051-79cd5c00-913f-4e76-8794-9d78c02ce326.png">
<img width="581" alt="Screenshot 2022-09-13 at 8 44 06 PM" src="https://user-images.githubusercontent.com/36603774/189940058-c9776ab3-d5df-4ae0-a2a2-479e229c9176.png">

The codegen file:
```
import {CodegenConfig} from '@graphql-codegen/cli';

const config: CodegenConfig = {
  schema: 'src/gql/schema.json',
  documents: [
    'src/**/*.tsx',
    'src/**/*.graphql',
    '!src/**/__tests__/*',
    '!src/**/*.spec.tsx',
    '!src/generated/**/*',
  ],
  generates: {
    './src/types-v3.ts': {
      plugins: [
        {
          add: {
            content: `
          /** THIS FILE IS AUTO-GENERATED **/
          /** DO NOT EDIT **/
          /* eslint-disable */
          /* tslint:disable */
          `,
          },
        },
        'typescript',
        'typescript-operations',
      ],
      config: {
        namingConvention: 'keep',
        avoidOptionals: {
          field: true,
        },
        scalars: {
          Long: 'number',
          UUID: 'string',
          DateTime: 'string',
          LocalTime: 'string',
          Duration: 'string',
          Void: 'void',
          URL: 'URL',
        },
        nonOptionalTypename: true,
        skipTypeNameForRoot: true,
        immutableTypes: true,
        declarationKind: 'interface',
        dedupeOperationSuffix: true,
        onlyOperationTypes: true,
      },
    },
  },
};

export default config;

```

**Comment by charlypoly** [comment]
We got a similar bug report (https://github.com/dotansimha/graphql-code-generator/discussions/8327), I will try to reproduce it and provide a patch.

**Comment by charlypoly** [comment, marked as answer]
Hi @nandita1,

Fields and scalars comments (called "descriptions" in GraphQL) should use the following syntax:

```graphql
"Link"
scalar URL

type Query {
    "Get a user"
    user(id: ID!): User!
}

type User {
    id: ID!
    "a username is unique"
    username: String
    email: String!
}
```

**Reply by charlypoly** [reply]
For more information, please refer to the following answer: https://github.com/dotansimha/graphql-code-generator/discussions/8327#discussioncomment-3707805
